File: java/src/main/java/com/experian/dvs/client/exceptions/UnauthorizedException.java
package com.experian.dvs.client.exceptions;

import com.experian.dvs.client.server.RestApiResponseError;

/**
 * Represents an exception that is thrown when a REST API call fails due to unauthorized access.
 */
public class UnauthorizedException extends EDVSException {

    /**
     * Initializes a new instance of the {@link UnauthorizedException} class with the details of the unauthorized error.
     *
     * @param responseError The error response from the REST API containing details about the unauthorized access.
     */
    public UnauthorizedException(final RestApiResponseError responseError) {
        super(responseError.getDetail());
    }
}

File: csharp/DVSClient/DVSClient/exceptions/UnauthorizedException.cs
using DVSClient.Server;

namespace DVSClient.Exceptions
{
    /// <summary>
    /// Represents an exception that is thrown when a REST API call fails due to unauthorized access.
    /// </summary>
    public class UnauthorizedException : EDVSException
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="UnauthorizedException"/> class with the details of the unauthorized error.
        /// </summary>
        /// <param name="responseError">The error response from the REST API containing details about the unauthorized access.</param>
        public UnauthorizedException(RestApiResponseError responseError) : base(responseError.Detail)
        {
        }
    }
}
